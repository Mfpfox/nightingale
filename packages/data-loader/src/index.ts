import { defineElement } from "@nightingale-elements/utils";
import DataLoader from "./data-loader";

defineElement(DataLoader);
