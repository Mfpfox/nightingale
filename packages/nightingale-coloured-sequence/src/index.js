import { defineElement } from "@nightingale-elements/utils";
import NightingaleColouredSequence from "./nightingale-coloured-sequence";

defineElement(NightingaleColouredSequence);

export default NightingaleColouredSequence;
