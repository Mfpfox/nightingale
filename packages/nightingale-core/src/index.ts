import NightingaleBaseElement from "./nightingale-base-element";
import _withZoom from "./withZoom";
import _withDimensions from "./withDimensions";
import _withPosition from "./withPosition";
import _withMargin from "./withMargin";

export const withZoom = _withZoom;
export const withDimensions = _withDimensions;
export const withPosition = _withPosition;
export const withMargin = _withMargin;

export default NightingaleBaseElement;
