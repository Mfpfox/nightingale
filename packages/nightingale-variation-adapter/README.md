# nightingale-navigation

A custom element allowing to navigate a protein/nucleic sequence.

## Usage

```html
<nightingale-variation-adapter></nightingale-variation-adapter>
```

## API Reference

### Properties

#### `name: type`

A property.

### Events

#### `my-event`

An event that does things.
