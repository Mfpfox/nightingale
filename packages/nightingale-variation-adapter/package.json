{
  "name": "@nightingale-elements/nightingale-variation-adapter",
  "version": "3.2.1",
  "description": "Data adapter for variation data",
  "files": [
    "dist",
    "types",
    "src"
  ],
  "main": "dist/nightingale-variation-adapter.js",
  "module": "dist/es/index.js",
  "types": "dist/es/nightingale-variation-adapter.d.ts",
  "repository": {
    "type": "git",
    "url": "https://github.com/ebi-webcomponents/nightingale.git"
  },
  "bugs": {
    "url": "https://github.com/ebi-webcomponents/nightingale/issues"
  },
  "homepage": "https://ebi-webcomponents.github.io/nightingale/",
  "keywords": [
    "nightingale",
    "customelements",
    "webcomponents",
    "variation"
  ],
  "author": "Xavier Watkins &lt;xwatkins@ebi.ac.uk&gt;",
  "license": "MIT",
  "gitHead": "337d88a3df2e52243aed81a123740d82917779db",
  "dependencies": {
    "@nightingale-elements/nightingale-feature-adapter": "^3.2.1",
    "@nightingale-elements/nightingale-variation": "^3.2.1",
    "@nightingale-elements/types": "^1.0.0",
    "@nightingale-elements/nightingale-core": "^1.0.0",
    "d3-scale": "^3.2.3",
    "lodash-es": "^4.17.11",
    "uuid": "^8.3.2"
  },
  "peerDependencies": {
    "d3": "5.X"
  },
  "devDependencies": {
    "@types/uuid": "^8.3.0"
  }
}
