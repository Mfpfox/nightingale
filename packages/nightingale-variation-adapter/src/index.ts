import { defineElement } from "@nightingale-elements/utils";
import NightingaleVariationAdapter from "./nightingale-variation-adapter";

defineElement(NightingaleVariationAdapter);

export default NightingaleVariationAdapter;
