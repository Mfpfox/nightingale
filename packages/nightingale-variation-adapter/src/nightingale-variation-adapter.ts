import NightingaleFeatureAdapter from "@nightingale-elements/nightingale-feature-adapter";
import { NightingaleVariationData } from "@nightingale-elements/nightingale-variation";
import { v1 } from "uuid";

import formatTooltip from "./tooltipGenerators";
import { ProteinsAPIVariation, Xref, SourceType } from "./variants";

const getSourceType = (xrefs: Xref[], sourceType: SourceType) => {
  const xrefNames = xrefs ? xrefs.map((ref) => ref.name) : [];
  if (sourceType === "uniprot" || sourceType === "mixed") {
    xrefNames.push("uniprot");
  }
  return xrefNames;
};

export const transformData = (
  data: ProteinsAPIVariation
): NightingaleVariationData => {
  const { sequence, features } = data;
  const variants = features.map((variant) => ({
    ...variant,
    accession: variant.genomicLocation,
    variant: variant.alternativeSequence ? variant.alternativeSequence : "-",
    start: variant.begin,
    xrefNames: getSourceType(variant.xrefs, variant.sourceType),
    hasPredictions: variant.predictions && variant.predictions.length > 0,
    tooltipContent: formatTooltip(variant),
    protvistaFeatureId: v1(),
  }));
  if (!variants) return null;
  return { sequence, variants };
};

class NightingaleVariationAdapter extends NightingaleFeatureAdapter {
  static is = "nightingale-variation-adapter";

  connectedCallback(): void {
    super.connectedCallback();
    if (this.closest("nightingale-manager")) {
      this.manager = this.closest("nightingale-manager");
      this.manager.register(this);
    }
  }

  disconnectedCallback(): void {
    if (this.manager) {
      this.manager.unregister(this);
    }
  }

  parseEntry(data: ProteinsAPIVariation): void {
    this._adaptedData = transformData(data);
  }
}

export default NightingaleVariationAdapter;
