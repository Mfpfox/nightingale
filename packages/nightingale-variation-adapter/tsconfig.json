{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "outDir": "./dist/es/",
    "declaration": true
  },
  "files": ["src/nightingale-variation-adapter.ts"]
}
