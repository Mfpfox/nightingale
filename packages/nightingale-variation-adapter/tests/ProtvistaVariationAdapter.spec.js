import { transformData } from "../src/nightingale-variation-adapter";
import variants from "./variants.json";
import {
  getDiseaseAssociations,
  getDescriptions,
  getPopulationFrequencies,
  getPredictions,
} from "../src/tooltipGenerators";

describe("NightingaleVariationAdapter", () => {
  it("should format disease associations", () => {
    const { features } = variants;
    const associations = getDiseaseAssociations(features[0].association);
    expect(associations).toMatchSnapshot();
  });

  it("should format descriptions", () => {
    const { features } = variants;
    const descriptions = getDescriptions(features[0].descriptions);
    expect(descriptions).toMatchSnapshot();
  });

  it("should format population frequencies", () => {
    const { features } = variants;
    const frequencies = getPopulationFrequencies(
      features[0].populationFrequencies
    );
    expect(frequencies).toMatchSnapshot();
  });

  it("should format predictions", () => {
    const { features } = variants;
    const predictions = getPredictions(features[0].predictions);
    expect(predictions).toMatchSnapshot();
  });

  it("should transform the data adequately", () => {
    const data = transformData(variants);
    expect(data).toMatchSnapshot();
  });
});
