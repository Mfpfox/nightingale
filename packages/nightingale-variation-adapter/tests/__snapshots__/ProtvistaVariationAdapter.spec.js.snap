// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NightingaleVariationAdapter should format descriptions 1`] = `
"<hr/><h5>Description</h5><p>Alzheimer disease</p>
  "
`;

exports[`NightingaleVariationAdapter should format disease associations 1`] = `
"
              <h4>Disease association</h4><p>Alzheimer disease (AD)</p>
              
              <h5>Cross-references</h5><ul><li style=\\"padding: .25rem 0\\">MIM <a href=\\"https://www.omim.org/entry/104300\\" style=\\"color:#FFF\\" target=\\"_blank\\">104300</a></li></ul>
              
      <ul><li title='Automatic assertion inferred from database entries' style=\\"padding: .25rem 0\\">Imported:&nbsp;20301340&nbsp;(<a href='https://www.ncbi.nlm.nih.gov/pubmed/20301340' style=\\"color:#FFF\\" target='_blank'>pubmed</a>&nbsp;<a href='https://europepmc.org/abstract/MED/20301340' style=\\"color:#FFF\\" target='_blank'>EuropePMC</a>)</li><li title='Automatic assertion inferred from database entries' style=\\"padding: .25rem 0\\">Imported:&nbsp;20301414&nbsp;(<a href='https://www.ncbi.nlm.nih.gov/pubmed/20301414' style=\\"color:#FFF\\" target='_blank'>pubmed</a>&nbsp;<a href='https://europepmc.org/abstract/MED/20301414' style=\\"color:#FFF\\" target='_blank'>EuropePMC</a>)</li><li title='Automatic assertion inferred from database entries' style=\\"padding: .25rem 0\\">Imported:&nbsp;20831773&nbsp;(<a href='https://www.ncbi.nlm.nih.gov/pubmed/20831773' style=\\"color:#FFF\\" target='_blank'>pubmed</a>&nbsp;<a href='https://europepmc.org/abstract/MED/20831773' style=\\"color:#FFF\\" target='_blank'>EuropePMC</a>)</li><li title='Automatic assertion inferred from database entries' style=\\"padding: .25rem 0\\">Imported:&nbsp;21500874&nbsp;(<a href='https://www.ncbi.nlm.nih.gov/pubmed/21500874' style=\\"color:#FFF\\" target='_blank'>pubmed</a>&nbsp;<a href='https://europepmc.org/abstract/MED/21500874' style=\\"color:#FFF\\" target='_blank'>EuropePMC</a>)</li><li title='Automatic assertion inferred from database entries' style=\\"padding: .25rem 0\\">Imported:&nbsp;24829003&nbsp;(<a href='https://www.ncbi.nlm.nih.gov/pubmed/24829003' style=\\"color:#FFF\\" target='_blank'>pubmed</a>&nbsp;<a href='https://europepmc.org/abstract/MED/24829003' style=\\"color:#FFF\\" target='_blank'>EuropePMC</a>)</li><li title='Automatic assertion inferred from database entries' style=\\"padding: .25rem 0\\">Imported:&nbsp;&nbsp;<a href='https://www.ncbi.nlm.nih.gov/clinvar/RCV000813633' style=\\"color:#FFF\\" target='_blank'>RCV000813633</a>&nbsp;(ClinVar)</li></ul>
    
          "
`;

exports[`NightingaleVariationAdapter should format population frequencies 1`] = `"<hr/><h5>Population frequencies</h5><p>0.0002 - MAF (1000Genomes)</p>"`;

exports[`NightingaleVariationAdapter should format predictions 1`] = `"<h6>PolyPhen</h6><ul class=\\"no-bullet\\"><li>probably damaging</li></ul>"`;

exports[`NightingaleVariationAdapter should transform the data adequately 1`] = `
Object {
  "sequence": "MLPGLALLLLAAWTARALEVPTDGNAGLLAEPQIAMFCGRLNMHMNVQNGKWDSDPSGTKTCIDTKEGILQYCQEVYPELQITNVVEANQPVTIQNWCKRGRKQCKTHPHFVIPYRCLVGEFVSDALLVPDKCKFLHQERMDVCETHLHWHTVAKETCSEKSTNLHDYGMLLPCGIDKFRGVEFVCCPLAEESDNVDSADAEEDDSDVWWGGADTDYADGSEDKVVEVAEEEEVAEVEEEEADDDEDDEDGDEVEEEAEEPYEEATERTTSIATTTTTTTESVEEVVREVCSEQAETGPCRAMISRWYFDVTEGKCAPFFYGGCGGNRNNFDTEEYCMAVCGSAMSQSLLKTTQEPLARDPVKLPTTAASTPDAVDKYLETPGDENEHAHFQKAKERLEAKHRERMSQVMREWEEAERQAKNLPKADKKAVIQHFQEKVESLEQEAANERQQLVETHMARVEAMLNDRRRLALENYITALQAVPPRPRHVFNMLKKYVRAEQKDRQHTLKHFEHVRMVDPKKAAQIRSQVMTHLRVIYERMNQSLSLLYNVPAVAEEIQDEVDELLQKEQNYSDDVLANMISEPRISYGNDALMPSLTETKTTVELLPVNGEFSLDDLQPWHSFGADSVPANTENEVEPVDARPAADRGLTTRPGSGLTNIKTEEISEVKMDAEFRHDSGYEVHHQKLVFFAEDVGSNKGAIIGLMVGGVVIATVIVITLVMLKKKQYTSIHHGVVEVDAAVTPEERHLSKMQQNGYENPTYKFFEQMQN",
  "variants": Array [
    Object {
      "accession": "NC_000021.9:g.26112051C>A",
      "alternativeSequence": "N",
      "association": Array [
        Object {
          "dbReferences": Array [
            Object {
              "id": "104300",
              "name": "MIM",
              "url": "https://www.omim.org/entry/104300",
            },
          ],
          "disease": true,
          "evidences": Array [
            Object {
              "code": "ECO:0000313",
              "source": Object {
                "alternativeUrl": "https://europepmc.org/abstract/MED/20301340",
                "id": "20301340",
                "name": "pubmed",
                "url": "https://www.ncbi.nlm.nih.gov/pubmed/20301340",
              },
            },
            Object {
              "code": "ECO:0000313",
              "source": Object {
                "alternativeUrl": "https://europepmc.org/abstract/MED/20301414",
                "id": "20301414",
                "name": "pubmed",
                "url": "https://www.ncbi.nlm.nih.gov/pubmed/20301414",
              },
            },
            Object {
              "code": "ECO:0000313",
              "source": Object {
                "alternativeUrl": "https://europepmc.org/abstract/MED/20831773",
                "id": "20831773",
                "name": "pubmed",
                "url": "https://www.ncbi.nlm.nih.gov/pubmed/20831773",
              },
            },
            Object {
              "code": "ECO:0000313",
              "source": Object {
                "alternativeUrl": "https://europepmc.org/abstract/MED/21500874",
                "id": "21500874",
                "name": "pubmed",
                "url": "https://www.ncbi.nlm.nih.gov/pubmed/21500874",
              },
            },
            Object {
              "code": "ECO:0000313",
              "source": Object {
                "alternativeUrl": "https://europepmc.org/abstract/MED/24829003",
                "id": "24829003",
                "name": "pubmed",
                "url": "https://www.ncbi.nlm.nih.gov/pubmed/24829003",
              },
            },
            Object {
              "code": "ECO:0000313",
              "source": Object {
                "alternativeUrl": "https://www.ensembl.org/homo_sapiens/Variation/Explore?v=RCV000813633",
                "id": "RCV000813633",
                "name": "ClinVar",
                "url": "https://www.ncbi.nlm.nih.gov/clinvar/RCV000813633",
              },
            },
          ],
          "name": "Alzheimer disease (AD)",
        },
      ],
      "begin": "51",
      "clinicalSignificances": Array [
        Object {
          "sources": Array [
            "ClinVar",
          ],
          "type": "Variant of uncertain significance",
        },
      ],
      "codon": "AAG/AAT",
      "consequenceType": "missense",
      "cytogeneticBand": "21q21.3",
      "descriptions": Array [
        Object {
          "sources": Array [
            "ClinVar",
          ],
          "value": "Alzheimer disease",
        },
      ],
      "end": "51",
      "genomicLocation": "NC_000021.9:g.26112051C>A",
      "hasPredictions": true,
      "locations": Array [
        Object {
          "loc": "p.Lys51Asn",
          "seqId": "ENST00000346798",
          "source": "Ensembl",
        },
        Object {
          "loc": "c.153G>T",
          "seqId": "ENST00000346798",
          "source": "Ensembl",
        },
      ],
      "mutatedType": "N",
      "populationFrequencies": Array [
        Object {
          "frequency": 0.0002,
          "populationName": "MAF",
          "source": "1000Genomes",
        },
      ],
      "predictions": Array [
        Object {
          "predAlgorithmNameType": "PolyPhen",
          "predictionValType": "probably damaging",
          "predictorType": "multi coding",
          "score": 0.983,
          "sources": Array [
            "Ensembl",
          ],
        },
        Object {
          "predAlgorithmNameType": "PolyPhen",
          "predictionValType": "probably damaging",
          "predictorType": "multi coding",
          "score": 0.954,
          "sources": Array [
            "Ensembl",
          ],
        },
      ],
      "protvistaFeatureId": "uuid_id",
      "somaticStatus": 0,
      "sourceType": "large_scale_study",
      "start": "51",
      "tooltipContent": "
<h5>Variant</h5><p>K > N</p>
<hr/><h5>Population frequencies</h5><p>0.0002 - MAF (1000Genomes)</p>
<h5>Consequence</h5><p>missense</p>

<h5>Location</h5><p>NC_000021.9:g.26112051C>A</p>

<hr/><h5>Description</h5><p>Alzheimer disease</p>
  

              <h4>Disease association</h4><p>Alzheimer disease (AD)</p>
              
              <h5>Cross-references</h5><ul><li style=\\"padding: .25rem 0\\">MIM <a href=\\"https://www.omim.org/entry/104300\\" style=\\"color:#FFF\\" target=\\"_blank\\">104300</a></li></ul>
              
      <ul><li title='Automatic assertion inferred from database entries' style=\\"padding: .25rem 0\\">Imported:&nbsp;20301340&nbsp;(<a href='https://www.ncbi.nlm.nih.gov/pubmed/20301340' style=\\"color:#FFF\\" target='_blank'>pubmed</a>&nbsp;<a href='https://europepmc.org/abstract/MED/20301340' style=\\"color:#FFF\\" target='_blank'>EuropePMC</a>)</li><li title='Automatic assertion inferred from database entries' style=\\"padding: .25rem 0\\">Imported:&nbsp;20301414&nbsp;(<a href='https://www.ncbi.nlm.nih.gov/pubmed/20301414' style=\\"color:#FFF\\" target='_blank'>pubmed</a>&nbsp;<a href='https://europepmc.org/abstract/MED/20301414' style=\\"color:#FFF\\" target='_blank'>EuropePMC</a>)</li><li title='Automatic assertion inferred from database entries' style=\\"padding: .25rem 0\\">Imported:&nbsp;20831773&nbsp;(<a href='https://www.ncbi.nlm.nih.gov/pubmed/20831773' style=\\"color:#FFF\\" target='_blank'>pubmed</a>&nbsp;<a href='https://europepmc.org/abstract/MED/20831773' style=\\"color:#FFF\\" target='_blank'>EuropePMC</a>)</li><li title='Automatic assertion inferred from database entries' style=\\"padding: .25rem 0\\">Imported:&nbsp;21500874&nbsp;(<a href='https://www.ncbi.nlm.nih.gov/pubmed/21500874' style=\\"color:#FFF\\" target='_blank'>pubmed</a>&nbsp;<a href='https://europepmc.org/abstract/MED/21500874' style=\\"color:#FFF\\" target='_blank'>EuropePMC</a>)</li><li title='Automatic assertion inferred from database entries' style=\\"padding: .25rem 0\\">Imported:&nbsp;24829003&nbsp;(<a href='https://www.ncbi.nlm.nih.gov/pubmed/24829003' style=\\"color:#FFF\\" target='_blank'>pubmed</a>&nbsp;<a href='https://europepmc.org/abstract/MED/24829003' style=\\"color:#FFF\\" target='_blank'>EuropePMC</a>)</li><li title='Automatic assertion inferred from database entries' style=\\"padding: .25rem 0\\">Imported:&nbsp;&nbsp;<a href='https://www.ncbi.nlm.nih.gov/clinvar/RCV000813633' style=\\"color:#FFF\\" target='_blank'>RCV000813633</a>&nbsp;(ClinVar)</li></ul>
    
          
<h6>PolyPhen</h6><ul class=\\"no-bullet\\"><li>probably damaging</li></ul>",
      "type": "VARIANT",
      "variant": "N",
      "wildType": "K",
      "xrefNames": Array [
        "1000Genomes",
        "ExAC",
        "gnomAD",
      ],
      "xrefs": Array [
        Object {
          "id": "rs529782627",
          "name": "1000Genomes",
          "url": "https://www.ensembl.org/homo_sapiens/Variation/Explore?v=rs529782627",
        },
        Object {
          "id": "rs529782627",
          "name": "ExAC",
          "url": "https://www.ncbi.nlm.nih.gov/SNP/snp_ref.cgi?type=rs&rs=rs529782627",
        },
        Object {
          "id": "rs529782627",
          "name": "gnomAD",
          "url": " http://gnomad.broadinstitute.org/awesome?query=rs529782627",
        },
      ],
    },
  ],
}
`;
